### ISSUE TITLE
Core dump on `pkgin fug` after installing third-party pckage

### ISSUE THREAD

**Issue by mcamou** [TritonDataCenter/pkgsrc-legacy#224]
After installing http://s3.amazonaws.com/downloads.basho.com/riak/2.0/2.0.2/smartos/13.1/riak-2.0.2-SmartOS-x86_64.tgz through `pkg_add`, doing a `pkgin fug` dumps core. I've generated the following trace and debug outputs:
- https://us-east.manta.joyent.com/yetu/public/pkgin-coredump/core.pkgin.18254.1415639497 - The core file
- https://us-east.manta.joyent.com/yetu/public/pkgin-coredump/pkgin.db - The pkgin.db file after the package was installed
- https://us-east.manta.joyent.com/yetu/public/pkgin-coredump/mdb_trace - The `mdb` output after generating the core
- https://us-east.manta.joyent.com/yetu/public/pkgin-coredump/truss.out - The `truss` output of running `pkgin fug`

More debug info:

```
# cat /etc/product
Name: Joyent Instance
Image: base64 14.2.0
Documentation: http://wiki.joyent.com/jpc2/SmartMachine+Base
# pkgin -v
pkgin 0.6.4 for SunOS-5.11 x86_64 (using SQLite 3.8.5)
```

/CC @bixu

**Comment by mamash** [comment]
There's probably something about the package that pkgin doesn't like. Unless they changed the workflow, Basho has been hacking the packages together ever since and using outdated pkgsrc infrastructure to do it too. I believe they are still using 2013Q1 package bits to test/release against.

We're working on updating our proper riak packages to 2.x in the meantime.

**Comment by bahamat** [comment]
Duplicate of #183.

This report demonstrates that the issue isn't the extra repo, but the non-indexed packages.

**Comment by mamash** [comment]
This should be fixed in upstream pkgin via NetBSDfr/pkgin#1.

**Comment by bahamat** [comment]
New images released today (14.4.0) include pkgin 0.8.0 which resolves this issue.
